libraries/SDFS/src/SDFS.h

```diff
@@ -0,0 +1,404 @@
+        strcpy(_name.get(), name);
+    }
+
+    ~SDFSFileImpl() override
+    {
+        flush();
+        close();
+    }
+
+    size_t write(const uint8_t *buf, size_t size) override
+    {
+        return _opened ? _fd->write(buf, size) : -1;
+    }
+
+    size_t read(uint8_t* buf, size_t size) override
+    {
+        return _opened ? _fd->read(buf, size) : -1;
+    }
+
+    void flush() override
+    {
+        if (_opened) {
+            _fd->flush();
+            _fd->sync();
+        }
+    }
+
+    bool seek(uint32_t pos, SeekMode mode) override
+    {
+        if (!_opened) {
+            return false;
+        }
+        switch (mode) {
+            case SeekSet:
+                return _fd->seekSet(pos);
+            case SeekEnd:
+                return _fd->seekEnd(-pos); // TODO again, odd from POSIX
+            case SeekCur:
+                return _fd->seekCur(pos);
+        }
+        return false;
+    }
+
+    size_t position() const override
+    {
+        return _opened ? _fd->curPosition() : 0;
+    }
+
+    size_t size() const override
+    {
+        return _opened ? _fd->fileSize() : 0;
+    }
+
+    bool truncate(uint32_t size) override
+    {
+        if (!_opened) {
+            return false;
+        }
+        return _fd->truncate(size);
+    }
+
+    void close() override
+    {
+        if (_opened) {
+            _fd->close();
+            _opened = false;
+        }
+    }
+
+    const char* name() const override
+    {
+        if (!_opened) {
+            return nullptr;
+        } else {
+            const char *p = _name.get();
+            const char *slash = strrchr(p, '/');
+            return (slash && slash[1]) ? slash + 1 : p;
+        }
+    }
+
+    const char* fullName() const override
+    {
+        return _opened ? _name.get() : nullptr;
+    }
+
+    bool isFile() const override
+    {
+        return _opened ? _fd->isFile() : false;;
+    }
+
+    bool isDirectory() const override
+    {
+        return _opened ? _fd->isDirectory() : false;
+    }
+
+
+protected:
+    SDFSImpl*                     _fs;
+    std::shared_ptr<sdfat::File>  _fd;
+    std::shared_ptr<char>         _name;
+    bool                          _opened;
+};
+
+class SDFSDirImpl : public DirImpl
+{
+public:
+    SDFSDirImpl(const String& pattern, SDFSImpl* fs, std::shared_ptr<sdfat::File> dir, const char *dirPath = nullptr)
+        : _pattern(pattern), _fs(fs), _dir(dir), _valid(false), _dirPath(nullptr)
+    {
+        if (dirPath) {
+            _dirPath = std::shared_ptr<char>(new char[strlen(dirPath) + 1], std::default_delete<char[]>());
+            strcpy(_dirPath.get(), dirPath);
+        }
+    }
+
+    ~SDFSDirImpl() override
+    {
+        _dir->close();
+    }
+
+    FileImplPtr openFile(OpenMode openMode, AccessMode accessMode) override
+    {
+        if (!_valid) {
+            return FileImplPtr();
+        }
+	char tmpName[128];
```

Comment:
Is 128 really ok? This looks like a limitation of the implementation. If so, it should be documented.

Reply:
Bumped to 260 bytes which is what Wikipedia has to say on the issue, with comment.  The SDFat lib doesn't really care, since the app (us) has to pass it in and keep track of it, not the lib.